opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -171,30 +172,30 @@ def __init__(
         self._metric_readers = metric_readers
 
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader._register_measurement_consumer(self)
+            metric_reader._set_collect_callback(
+                self._measurement_consumer.collect
+            )
 
+        self._shutdown_once = Once()
         self._shutdown = False
 
     def force_flush(self) -> bool:
 
         # FIXME implement a timeout
 
-        metric_reader_result = True
-
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader_result = (
-                metric_reader_result and metric_reader.force_flush()
-            )
-
-        if not metric_reader_result:
```

Comment:
I might've missed the discussion for this, why are we removing the log messages for failed attempts?

Reply:
Metric reader interface doesn't have `force_flush`. The direct call to `collect` basically does that and it doesn't return anything. There is #2402 to implement the timeout which will take of this.